app/scripts/lib/AccountIdentitiesPetnamesBridge.test.ts

```diff
@@ -129,51 +162,80 @@ function createNameControllerMock(
 }
 
 describe('AccountIdentitiesPetnamesBridge', () => {
+  let messenger;
+  let accountsControllerMessenger: AccountsControllerMessenger;
+  let petnamesBridgeMessenger: PetnamesBridgeMessenger<
+    AccountIdentitiesPetnamesBridgeEvents,
+    AccountIdentitiesPetnamesBridgeActions
+  >;
+
   beforeEach(() => {
+    messenger = new ControllerMessenger();
```

Comment:
I'm a big fan of isolating unit tests as much as possible by mocking anything we don't want to test.

We're using a real instances of the controller messenger meaning if that code fails, these tests now fail also.

In this scenario, we can simulate the messenger events by provided an entirely mocked `messenger` object to the constructor and then just invoking the first argument passed to the `subscribe` method as that's the limit of our interaction in the implementation.

Reply:
@montelaidev – check out the way the messenger is mocked in AddressBookPetnamesBridge.test.ts – and how the first argument, the listener callback, is invoked:
https://github.com/MetaMask/metamask-extension/blob/b16ad3171e7c2b36d611f23c43b5ebb532fdfadd/app/scripts/lib/AddressBookPetnamesBridge.test.ts#L138-L140

Reply:
That makes sense, i've changed the tests to use a mocked messenger